src/game/Spell.cpp

```diff
@@ -4193,6 +4231,9 @@ SpellCastResult Spell::CheckCast(bool strict)
             return SPELL_FAILED_NOT_READY;
     }
 
+    if (!m_caster->isAlive() && m_caster->GetTypeId() == TYPEID_PLAYER && !m_spellInfo->HasAttribute(SPELL_ATTR_CASTABLE_WHILE_DEAD))
```

Comment:
It's only 1 spared logical gate so I guess you're right it makes no real difference on performance. Just a design choise I guess =)